pkg/apis/externaldns/types.go

```diff
@@ -235,8 +231,6 @@ func (cfg *Config) ParseFlags(args []string) error {
 	app.Flag("publish-internal-services", "Allow external-dns to publish DNS records for ClusterIP services (optional)").BoolVar(&cfg.PublishInternal)
 	app.Flag("publish-host-ip", "Allow external-dns to publish host-ip for headless services (optional)").BoolVar(&cfg.PublishHostIP)
 	app.Flag("connector-source-server", "The server to connect for connector source, valid only when using connector source").Default(defaultConfig.ConnectorSourceServer).StringVar(&cfg.ConnectorSourceServer)
-	app.Flag("crd-source-apiversion", "API version of the CRD for crd source, e.g. `externaldns.k8s.io/v1alpha1`, valid only when using crd source").Default(defaultConfig.CRDSourceAPIVersion).StringVar(&cfg.CRDSourceAPIVersion)
```

Comment:
Why this flags are removed. What is the motivation behind removing them?

This flags are the glue to some 3rd party components who would generate the DNS records. `external-dns ` would watch (list in current implementation) them and take necessary action. without these flags, now the name and group of the CRD is hard coded.

Reply:
To be honest I didn't really understand why we might want to ask users for this input as opposed to external-dns defining it's CRDs and versions that they should use. I feel like that makes more sense long term, but I get that removing the flags is a breaking change, so I guess we can keep them for now and look to deprecate later ?